src/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-revisions-controller.php

```diff
@@ -0,0 +1,484 @@
+<?php
+/**
+ * REST API: WP_REST_Global_Styles_Revisions_Controller class
+ *
+ * @package WordPress
+ * @subpackage REST_API
+ * @since 6.3.0
+ */
+
+/**
+ * Core class used to access global styles revisions via the REST API.
+ *
+ * @since 6.3.0
+ *
+ * @see WP_REST_Controller
+ */
+class WP_REST_Global_Styles_Revisions_Controller extends WP_REST_Controller {
+	/**
+	 * Parent post type.
+	 *
+	 * @since 6.3.0
+	 * @var string
+	 */
+	private $parent_post_type;
+
+	/**
+	 * The base of the parent controller's route.
+	 *
+	 * @since 6.3.0
+	 * @var string
+	 */
+	private $parent_base;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 6.3.0
+	 */
+	public function __construct() {
+		$this->parent_post_type = 'wp_global_styles';
+		$this->rest_base        = 'revisions';
+		$this->parent_base      = 'global-styles';
+		$this->namespace        = 'wp/v2';
+	}
+
+	/**
+	 * Registers the controllers routes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->parent_base . '/(?P<parent>[\d]+)/' . $this->rest_base,
+			array(
+				'args'   => array(
+					'parent' => array(
+						'description' => __( 'The ID for the parent of the revision.' ),
+						'type'        => 'integer',
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_item_permissions_check' ),
+					'args'                => $this->get_collection_params(),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Retrieves the query params for collections.
+	 * Taken mostly from WP_REST_Controller->get_collection_params().
+	 *
+	 * @since 6.3.0
+	 *
+	 * @return array Collection parameters.
+	 */
+	public function get_collection_params() {
+		return array(
+			'context'  => array( 'default' => 'view' ),
+			'page'     => array(
+				'description'       => __( 'Current page of the collection.' ),
+				'type'              => 'integer',
+				'default'           => 1,
+				'sanitize_callback' => 'absint',
+				'validate_callback' => 'rest_validate_request_arg',
+				'minimum'           => 1,
+			),
+			'per_page' => array(
+				'description'       => __( 'Maximum number of items to be returned in result set.' ),
+				'type'              => 'integer',
+				'minimum'           => 1,
+				'maximum'           => 100,
+				'sanitize_callback' => 'absint',
+				'validate_callback' => 'rest_validate_request_arg',
+			),
+			'offset'   => array(
+				'description' => __( 'Offset the result set by a specific number of items.' ),
+				'type'        => 'integer',
+			),
+		);
```

Comment:
```suggestion
		$collection_params = $this->get_collection_params();
		unset( $collection_params['search'] );
		
		return $collection_params;
```

This might be a little cleaner. And less translated strings.

Reply:
Looks good! I think I'll still need to add `offset` since it's not inheriting from `WP_REST_Revisions_Controller`